apps/dialog/src/routes/-components/Permissions.tsx

```diff
@@ -0,0 +1,178 @@
+  }, [limit, decimals.data, token])
+
+  return (
+    <div className="flex items-center gap-2 py-3 text-secondary text-sm">
+      <div className="flex h-8 w-8 items-center justify-center rounded-full bg-surface">
+        <LucidePiggyBank className="h-5 w-5" />
+      </div>
+      {displayAmount || !token ? (
+        <>
+          Spend {displayAmount} {symbol.data ?? 'ETH'} per {period}
+        </>
+      ) : (
+        <Spinner className="text-secondary" />
+      )}
+    </div>
+  )
+}
+
+function ContractAccessPermission(props: ContractAccessPermission.Props) {
+  const { calls } = props
+  const [isOpen, setIsOpen] = useState(false)
+
+  return (
+    <div className="flex flex-col">
+      <button
+        className="flex items-center gap-2 py-3 text-secondary text-sm"
+        onClick={() => setIsOpen(!isOpen)}
+        type="button"
+      >
+        <div className="flex h-8 w-8 items-center justify-center rounded-full bg-surface">
+          <LucideKey className="h-5 w-5" />
+        </div>
+        <span className="flex-1 text-left">Access-related permissions</span>
+        {isOpen ? (
+          <LucideChevronUp className="h-4 w-4" />
+        ) : (
+          <LucideChevronDown className="h-4 w-4" />
+        )}
+      </button>
+      {isOpen && (
+        <div className="space-y-2 pl-2">
+          <div className="flex items-center font-medium text-secondary text-sm">
+            <div className="flex-1 pl-8">
+              <span className="tracking-wider">Contract</span>
+            </div>
+            <div className="w-[8.75rem]">
+              <span className="tracking-wider">Function</span>
+            </div>
+          </div>
+          {calls.map((call) => (
+            <div
+              className="flex items-center text-secondary text-sm"
+              key={`call-${call.signature}-${call.to}`}
+            >
+              <div className="flex flex-1 items-center">
+                <div className="flex h-6 w-6 items-center justify-center rounded-full bg-emerald-100">
+                  <WalletIcon className="h-4 w-4 text-teal-600" />
+                </div>
+                <span className="ml-2 font-mono text-xs">
+                  {call.to ? StringFormatter.truncate(call.to) : 'Any contract'}
+                </span>
+              </div>
+              <div className="w-[8.75rem] font-mono">
+                {call.signature || 'Any function'}
+              </div>
+            </div>
+          ))}
+        </div>
+      )}
+    </div>
+  )
+}
+
+declare namespace Permissions {
+  type Props = {
+    capabilities?: {
+      grantPermissions?: {
+        permissions: {
+          spend?: readonly {
+            limit: `0x${string}`
+            period: 'minute' | 'hour' | 'day' | 'week' | 'month' | 'year'
+            token?: `0x${string}`
+          }[]
+          calls?: readonly {
+            signature?: string
+            to?: `0x${string}`
+          }[]
+        }
+      }
+    }
+  }
```

Comment:
What if we made this type more simpler? Then up in `wallet_connect.tsx`, we just pass through `permissions={capabilities?.grantPermissions?.permissions}` as a prop to `SignIn` & `SignUp`? This should make usage (e.g. L17-18) cleaner too.

```suggestion
          spend?: readonly SpendPermission.Props[]
          calls?: ContractAccessPermission.Props['calls']
  }
```